src/Handler/MailCryptKeyHandler.php

```diff
@@ -61,69 +61,51 @@ public function toPkcs8(string $privateKey): string
     /**
      * @throws Exception
      */
-    public function create(User $user): void
+    public function create(User $user, string $password): void
     {
         $pKey = openssl_pkey_new([
             'private_key_type' => self::MAIL_CRYPT_PRIVATE_KEY_TYPE,
             'curve_name' => self::MAIL_CRYPT_CURVE_NAME,
         ]);
         openssl_pkey_export($pKey, $privateKey);
         $privateKey = base64_encode($this->toPkcs8($privateKey));
-        $keyPair = new MailCryptKeyPair($privateKey, base64_encode((string) openssl_pkey_get_details($pKey)['key']));
+        $keyPair = new MailCryptKeyPair($privateKey, base64_encode((string)openssl_pkey_get_details($pKey)['key']));
         sodium_memzero($privateKey);
 
-        // get plain user password
-        if (null === $plainPassword = $user->getPlainPassword()) {
-            throw new Exception('plainPassword should not be null');
-        }
-
-        $mailCryptSecretBox = CryptoSecretHandler::create($keyPair->getPrivateKey(), $plainPassword);
+        $mailCryptSecretBox = CryptoSecretHandler::create($keyPair->getPrivateKey(), $password);
         $user->setMailCryptPublicKey($keyPair->getPublicKey());
         $user->setMailCryptSecretBox($mailCryptSecretBox->encode());
         $user->setPlainMailCryptPrivateKey($keyPair->getPrivateKey());
 
         // Clear variables with confidential content from memory
         $keyPair->erase();
-        sodium_memzero($plainPassword);
```

Comment:
I don't know how to do it better, but this looks like a bit of a regression. @doobry-systemli ?